packages/analytics-remote-config/src/remote-config-idb-store.ts

```diff
@@ -0,0 +1,87 @@
+import { Logger as ILogger } from '@amplitude/analytics-types';
+import { DBSchema, openDB } from 'idb';
+import {
+  ConfigNamespace,
+  RemoteConfig,
+  RemoteConfigAPIResponse,
+  RemoteConfigIDBStore,
+  SessionReplayRemoteConfig,
+} from './types';
+
+export interface RemoteConfigDB extends DBSchema {
+  [ConfigNamespace.SESSION_REPLAY]: {
```

Comment:
If we are making this generic, shouldn't the namespace be passed as a parameter to this class?

Reply:
Agreed, I was thinking that the Remote config service would focus on loading a generic config (i.e. key value pairs) and then provide a callback for when the config has been updated. 
The SDK/plugins would be responsible for copying the loaded values (by generic keys) to it's own strongly typed local config.

Reply:
Yeah agreed with this, up until the definition of the IndexedDB store itself. I chose to have this package manage the IndexedDB store, rather than have each consumer manage their own (which would be pretty repetitive). I'm very hesitant to provide a generic interface to the definition of the IndexedDB store, as IndexedDB is a database that requires versioning when making updates, so I think we could run into trouble generalizing it.

Instead of thinking of this as wholly generic, I'm thinking of it as the client match to the [Remote Config Service](https://amplitude.atlassian.net/wiki/spaces/IG/pages/2412806453/Remote+Config+Service+Interface+Design+WIP). As a service, it knows about the existence of certain keys and schemas, and provides the typing for those.

Reply:
I think I do agree with trying to make this a bit more generic. While I'm not certain how to provide a clean interface yet, I think it'll be easier to manage in the long run. This package provides a good amount of help with interfacing with the service and providing a caching mechanism.

However, if it starts to know more about the configs and we have stronger typing in this repo when new teams onboard they would need to:
1. Plan the config schema for their service and put that in the remote config service.
2. Come to this repo and adding the translation layer + typing. Hopefully they wouldn't need to know details of the caching mechanism (IndexedDB), since that adds room for error.
3. Change in their plugin/package/etc to utilize this new version with the new strict typing.

Now that would lead to this package being updated every time there is a model change, which can be fine, but seems like additional overhead if they're the only one using that new model.

Alternatively, if this could just focus on querying the service, providing a caching layer, and returning a generic object that the client would need to parse and strongly type. This way this package changes less often and clients of this package could operate independently without having two update two packages (this one and their own).

Reply:
> Alternatively, if this could just focus on querying the service, providing a caching layer, and returning a generic object that the client would need to parse and strongly type

I was kinda leaning towards what @lewgordon-amplitude mentioned too. 

Can the storage layer be something like this?
```
Namespace | Config json | sessionId
```

The remote config store takes in the namespace, version and session id as input, checks if the data is already exists for this combination and returns it. If not, fetches the latest data, persists it and returns it.

The remote config data is ephemeral. We should never have to do a migration. We can pretty much nuke it and fetch from the remote service again if there's any change in the schema.

Reply:
But @kwalker3690 , pretty sure you've thought about this and since you've been in the thick of the implementation, let us know if there's something that we are missing.

Reply:
If a new namespace comes in, the caller needs to pass in the new namespace and this component can construct the API to call and fetch this data and store it in indexdb.